src/main.rs

```diff
@@ -0,0 +1,124 @@
+// Require use of WinMain entrypoint so cmd doesn't flash open
+// See RFC 1665 for additional details
+#![windows_subsystem = "windows"]
+
+use winreg::RegKey;
+use std::process::{Command, exit};
+use std::io::Result;
+use std::env;
+use native_dialog::MessageDialog;
+use native_dialog::MessageType;
+use webbrowser;
+
+fn main() {
+    if let Ok(dir) = get_minecraft_installation_dir() {
+        println!("Minecraft install dir {}", dir);
+        try_minecraft_java(&dir);
+    } else {
+        println!("Could not find minecraft install dir.");
+    }
+
+    match env::var("JAVA_HOME") {
+        Ok(val) => {
+            let full_path: String = val.to_owned() + r"bin\javaw.exe";
```

Comment:
The `to_owned()` isn't needed?

Reason being you already have an owned String and you are converting to owned form again